src/ray/raylet/node_manager.cc

```diff
@@ -2969,10 +2969,14 @@ MemoryUsageRefreshCallback NodeManager::CreateMemoryUsageRefreshCallback() {
           /// TODO: (clarng) right now destroy is called after the messages are created
           /// since we print the process memory in the message. Destroy should be called
           /// as soon as possible to free up memory.
-          DestroyWorker(high_memory_eviction_target_,
-                        rpc::WorkerExitType::USER_ERROR,
-                        worker_exit_message,
-                        true /* force */);
+          this->io_service_.post(
```

Comment:
Why don't we put the whole code into this service? There are still potential races. E.g., accessing `        auto workers = this->WorkersWithLatestSubmittedTasks();
`

Reply:
agree - looks like the subcomponents of the node manager are sharing the event loop, seems more consistent to have the memory monitor be the same